### ISSUE TITLE
Failure in Vector3Interop_ro_cmd

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#21935]
For x64, JitStress=1/2.

https://ci.dot.net/job/dotnet_coreclr/job/master/view/x64/job/jitstress/job/x64_checked_windows_nt_jitstress1/738/
https://ci.dot.net/job/dotnet_coreclr/job/master/view/x64/job/jitstress/job/x64_checked_windows_nt_jitstress2/728/

Looks like a GC hole.

```
19:01:32 
      JIT_SIMD._Vector3Interop_ro_Vector3Interop_ro_._Vector3Interop_ro_Vector3Interop_ro_cmd [FAIL]
19:01:32         
19:01:32 
        Assert failure(PID 7652 [0x00001de4], Thread: 11872 [0x2e60]): !CREATE_CHECK_STRING(!"Detected use of a corrupted OBJECTREF. Possible GC hole.")
19:01:32         
19:01:32         CORECLR! `Object::ValidateInner'::`1'::catch$12 + 0x13E (0x00007ffc`9599dc54)
19:01:32         CORECLR! CallSettingFrame + 0x20 (0x00007ffc`9588cc50)
19:01:32         CORECLR! _FrameHandler3::CxxCallCatchBlock + 0x15A (0x00007ffc`9588b3fa)
19:01:32         NTDLL! RtlCaptureContext + 0x3C3 (0x00007ffc`b2122a63)
19:01:32         CORECLR! Object::ValidateInner + 0x73 (0x00007ffc`94fbd433)
19:01:32         CORECLR! Object::Validate + 0x13A (0x00007ffc`94fbd29a)
19:01:32         CORECLR! OBJECTREF::operator= + 0x28 (0x00007ffc`94fb54b8)
19:01:32         CORECLR! LayoutUpdateNative + 0x37A (0x00007ffc`95650ace)
19:01:32         CORECLR! FmtValueTypeUpdateNative + 0x39D (0x00007ffc`9564db39)
19:01:32         CORECLR! StubHelpers::ValueClassMarshaler__ConvertToNative + 0x1C9 (0x00007ffc`952cf929)
19:01:32             File: d:\j\workspace\x64_checked_w---02832afb\src\vm\object.cpp Line: 775
19:01:32             Image: D:\j\workspace\x64_checked_w---02832afb\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\CoreRun.exe
19:01:32         
19:01:32         
19:01:32   
19:01:32   Return code:      1
19:01:32   Raw output file:      D:\j\workspace\x64_checked_w---02832afb\bin\tests\Windows_NT.x64.Checked\Reports\JIT.SIMD\Vector3Interop_ro\Vector3Interop_ro.output.txt
19:01:32   Raw output:
19:01:32   BEGIN EXECUTION
19:01:32          "D:\j\workspace\x64_checked_w---02832afb\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\corerun.exe" Vector3Interop_ro.exe 
19:01:32         nativeCall_PInvoke_CheckVector3Size: sizeof(Vector3) == 12
19:01:32         nativeCall_PInvoke_Vector3Arg:
19:01:32             iVal 123
19:01:32             sumOfEles(1.000000, 2.000000, 3.000000) = 6.000000
19:01:32             str  abcdefg
19:01:32             sumOfEles(10.000000, 11.000000, 12.000000) = 33.000000
19:01:32         nativeCall_PInvoke_Vector3Arg_Unix:
19:01:32             v3f32_xmm0: 1.000000 2.000000 3.000000
19:01:32             f32_xmm2 - f32_xmm7: 100.000000 101.000000 102.000000 103.000000 104.000000 105.000000
19:01:32             f32_mem0: 106.000000
19:01:32             v3f32_mem1: 10.000000 20.000000 30.000000
19:01:32             f32_mem2-3: 107.000000 108.000000
19:01:32             sum = 1002.000000
19:01:32         nativeCall_PInvoke_Vector3Arg_Unix2:
19:01:32             v3f32_xmm0: 1.000000 2.000000 3.000000
19:01:32             f32_xmm2 - f32_xmm7: 100.000000 101.000000 102.000000 103.000000 104.000000 105.000000
19:01:32             f32_mem0: 106.000000
19:01:32             v3f32_mem1: 4.000000 5.000000 6.000000
19:01:32             f32_mem2-3: 107.000000 108.000000
19:01:32             v3f32_mem4: 7.000000 8.000000 9.000000
19:01:32             f32_mem5: 109.000000
19:01:32             sum = 1090.000000
19:01:32         nativeCall_PInvoke_Vector3Ret:
19:01:32             Return value: (1.000000, 2.000000, 3.000000)
19:01:32             Sum of return scalar values = 6.000000
19:01:32         nativeCall_PInvoke_Vector3Array
19:01:32             arrEle[0]: 1.000000 2.000000 3.000000
19:01:32             arrEle[1]: 5.000000 6.000000 7.000000
19:01:32             Sum = 24.000000
19:01:32         nativeCall_PInvoke_Vector3InStruct
19:01:32             First struct memeber: (1.000000 2.000000 3.000000) -> (2.000000 3.000000 4.000000)
19:01:32             Second struct member: (5.000000 6.000000 7.000000) -> (6.000000 7.000000 8.000000)
19:01:32             Sum of all return scalar values = 30.000000
19:01:32         Expected: 100
19:01:32         Actual: -1073740286
19:01:32         END EXECUTION - FAILED
19:01:32         FAILED
19:01:32         Test Harness Exitcode is : 1
19:01:32         
19:01:32   To run the test:
19:01:32   > set CORE_ROOT=D:\j\workspace\x64_checked_w---02832afb\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root
19:01:32   > D:\j\workspace\x64_checked_w---02832afb\bin\tests\Windows_NT.x64.Checked\JIT\SIMD\Vector3Interop_ro\Vector3Interop_ro.cmd
19:01:32   
19:01:32         Expected: True
19:01:32         Actual:   False
19:01:32         Stack Trace:
19:01:32           D:\j\workspace\x64_checked_w---02832afb\bin\tests\Windows_NT.x64.Checked\TestWrappers\JIT.SIMD\JIT.SIMD.XUnitWrapper.cs(3469,0): at JIT_SIMD._Vector3Interop_ro_Vector3Interop_ro_._Vector3Interop_ro_Vector3Interop_ro_cmd()
```

**Comment by BruceForstall** [comment]
@fiigii @tannergooding @CarolEidt @dotnet/jit-contrib

**Comment by BruceForstall** [comment]
Also seen on x86

**Comment by echesakovMSFT** [comment]
Also seen on Linux/arm64
```
export COMPlus_TieredCompilation=0
export COMPlus_JitStress=2
+ /ssd/dotnetbuild/work/fe7a2011-f1ba-4d09-84c2-0461957b32c9/Payload/corerun /ssd/dotnetbuild/work/fe7a2011-f1ba-4d09-84c2-0461957b32c9/Payload/xunit.console.dll JIT.SIMD.XUnitWrapper.dll -parallel collections -nocolor -noshadow -xml testResults.xml -notrait category=outerloop -notrait category=failing
xUnit.net Console Runner v2.4.1 (64-bit .NET Core 4.6.27310.0)
  Discovering: JIT.SIMD.XUnitWrapper
  Discovered:  JIT.SIMD.XUnitWrapper
  Starting:    JIT.SIMD.XUnitWrapper
    JIT_SIMD._Vector3Interop_ro_Vector3Interop_ro_._Vector3Interop_ro_Vector3Interop_ro_sh [FAIL]
      /ssd/dotnetbuild/work/fe7a2011-f1ba-4d09-84c2-0461957b32c9/Work/c0ff55d1-36b9-4724-bbd1-bc6b743ae518/Exec/Vector3Interop_ro/Vector3Interop_ro.sh: line 248: 32873 Segmentation fault      (core dumped) $_DebuggerFullPath "$CORE_ROOT/corerun" $ExePath $CLRTestExecutionArguments
      /n/nReturn code:      1/nRaw output file:      /ssd/dotnetbuild/work/fe7a2011-f1ba-4d09-84c2-0461957b32c9/Work/c0ff55d1-36b9-4724-bbd1-bc6b743ae518/Exec/Reports/JIT.SIMD/Vector3Interop_ro/Vector3Interop_ro.output.txt/nRaw output:/nBEGIN EXECUTION
      /ssd/dotnetbuild/work/fe7a2011-f1ba-4d09-84c2-0461957b32c9/Payload/corerun Vector3Interop_ro.exe
      nativeCall_PInvoke_CheckVector3Size: sizeof(Vector3) == 12
      nativeCall_PInvoke_Vector3Arg:
          iVal 123
          sumOfEles(1.000000, 2.000000, 3.000000) = 6.000000
          str  abcdefg
          sumOfEles(10.000000, 11.000000, 12.000000) = 33.000000
      nativeCall_PInvoke_Vector3Arg_Unix:
          v3f32_xmm0: 1.000000 2.000000 3.000000
          f32_xmm2 - f32_xmm7: 100.000000 101.000000 102.000000 103.000000 104.000000 105.000000
          f32_mem0: 106.000000
          v3f32_mem1: 10.000000 20.000000 30.000000
          f32_mem2-3: 107.000000 108.000000
          sum = 1002.000000
      nativeCall_PInvoke_Vector3Arg_Unix2:
          v3f32_xmm0: 1.000000 2.000000 3.000000
          f32_xmm2 - f32_xmm7: 100.000000 101.000000 102.000000 103.000000 104.000000 105.000000
          f32_mem0: 106.000000
          v3f32_mem1: 4.000000 5.000000 6.000000
          f32_mem2-3: 107.000000 108.000000
          v3f32_mem4: 7.000000 8.000000 9.000000
          f32_mem5: 109.000000
          sum = 1090.000000
      nativeCall_PInvoke_Vector3Ret:
          Return value: (1.000000, 2.000000, 3.000000)
          Sum of return scalar values = 6.000000
      nativeCall_PInvoke_Vector3Array
          arrEle[0]: 1.000000 2.000000 3.000000
          arrEle[1]: 5.000000 6.000000 7.000000
          Sum = 24.000000
      nativeCall_PInvoke_Vector3InStruct
          First struct memeber: (1.000000 2.000000 3.000000) -> (2.000000 3.000000 4.000000)
          Second struct member: (5.000000 6.000000 7.000000) -> (6.000000 7.000000 8.000000)
          Sum of all return scalar values = 30.000000
      Expected: 100
      Actual: 139
      END EXECUTION - FAILED
      Test Harness Exitcode is : 1
      /nTo run the test:/n> set CORE_ROOT=/ssd/dotnetbuild/work/fe7a2011-f1ba-4d09-84c2-0461957b32c9/Payload/n> /ssd/dotnetbuild/work/fe7a2011-f1ba-4d09-84c2-0461957b32c9/Work/c0ff55d1-36b9-4724-bbd1-bc6b743ae518/Exec/Vector3Interop_ro/Vector3Interop_ro.sh/n
      Expected: True
      Actual:   False
      Stack Trace:
        /__w/1/s/bin/tests/Linux.arm64.Checked/TestWrappers/JIT.SIMD/JIT.SIMD.XUnitWrapper.cs(3539,0): at JIT_SIMD._Vector3Interop_ro_Vector3Interop_ro_._Vector3Interop_ro_Vector3Interop_ro_sh()
  Finished:    JIT.SIMD.XUnitWrapper
=== TEST EXECUTION SUMMARY ===
   JIT.SIMD.XUnitWrapper  Total: 108, Errors: 0, Failed: 1, Skipped: 0, Time: 4.275s
```

**Comment by BruceForstall** [comment]
First Linux/arm64 failure:

https://ci.dot.net/job/dotnet_coreclr/job/master/view/arm64/job/jitstress/job/arm64_cross_checked_ubuntu16.04_jitstress1_flow/103/

Changes:
Changes
Delete SMALL_TREE_NODES (commit: d2297e7) (detail / githubweb)
Delete GenTreeJumpTable (commit: b94d2ea) (detail / githubweb)
Ifdef out GenTreeLclVar::gtLclILoffs (commit: d10a433) (detail / githubweb)
Delete GenTreeLabel (commit: 3d4b337) (detail / githubweb)
Check the optimize settings of JIT test projects. (commit: 1435bdf) (detail / githubweb)
Fixed the broken projects. (commit: efde54b) (detail / githubweb)
JIT: modify inline budget update to use estimated imported IL size (commit: 3f182cf) (detail / githubweb)
check for valid IL offsets before computing block ILsize (commit: c3a1320) (detail / githubweb)
Update gtLclILoffs related comments (commit: 4b68c59) (detail / githubweb)
fix DateTime overflow on ARM32 (#21902) (commit: 3c81b68) (detail / githubweb)
Moves common COM type to shared corelib partition (dotnet/corert#6782) (commit: c44b38f) (detail / githubweb)
Delete files moved to shared partition (commit: b2ea27c) (detail / githubweb)
Removes PNSE version of DispatchWrapper and keep only single version (commit: 006771e) (detail / githubweb)
More DispatchWrapper tweaks (commit: f76e33e) (detail / githubweb)
Support building with python3 on unix (#19356) (commit: 7e0608f) (detail / githubweb)
Remove extraneous eightbytes check for native structures and add tests. (commit: 03e2c02) (detail / githubweb)
Fix perf-prep to pass the branch name (#21911) (commit: 92daaff) (detail / githubweb)
Change CoreLib native image to be R2R by default on all platforms (commit: 5478a2f) (detail / githubweb)
Fix DynamicMethodDesc memory leak (#21891) (commit: 660eb00) (detail / githubweb)
Addressing https://github.com/dotnet/coreclr/issues/21461 (#21771) (commit: 6d00b98) (detail / githubweb)
Move OperationStatus into shared source (#21912) (commit: 5c480a4) (detail / githubweb)
Add tests for our layout-class marshalling (#20867) (commit: 9b3ace6) (detail / githubweb)
Specify the minimum supported version of macOS (#21816) (commit: fe5f8d6) (detail / githubweb)
Implement AssemblyDependencyResolver (#21896) (commit: dbf50d8) (detail / githubweb)
